apps/stripe/src/modules/app-config/file-app-config-presistor.ts

```diff
@@ -0,0 +1,237 @@
+
+  private persistExistingAppConfig(args: {
+    channelId: string;
+    config: StripeConfig;
+    saleorApiUrl: string;
+    appId: string;
+  }) {
+    const existingConfigResult = this.readExistingAppConfigFromFile();
+
+    if (existingConfigResult.isErr()) {
+      return err(existingConfigResult.error);
+    }
+
+    const newConfig: z.infer<typeof FileAppConfigPresistor.FileConfigSchema> = {
+      appRootConfig: {
+        ...existingConfigResult.value.appRootConfig,
+        [args.channelId]: this.serializeStripeConfigToJson(args.config),
+      },
+    };
+
+    const writeResult = this.writeJsonFileSafe(
+      this.filePath,
+      JSON.stringify(newConfig, null, 2),
+      "utf-8",
+    );
+
+    if (writeResult.isErr()) {
+      return err(writeResult.error);
+    }
+
+    return ok(undefined);
+  }
+
+  private persistNewAppConfig(args: {
+    channelId: string;
+    config: StripeConfig;
+    saleorApiUrl: string;
+    appId: string;
+  }) {
+    const newConfig = {
+      appConfig: {
+        [args.channelId]: this.serializeStripeConfigToJson(args.config),
+      },
+    };
+
+    const writeResult = this.writeJsonFileSafe(
+      this.filePath,
+      JSON.stringify(newConfig, null, 2),
+      "utf-8",
+    );
+
+    if (writeResult.isErr()) {
+      return err(writeResult.error);
+    }
+
+    return ok(undefined);
+  }
+
+  async saveStripeConfig(args: {
+    channelId: string;
+    config: StripeConfig;
+    saleorApiUrl: string;
+    appId: string;
+  }): Promise<Result<void, InstanceType<typeof BaseError>>> {
+    if (fs.existsSync(this.filePath)) {
+      return this.persistExistingAppConfig(args);
+    }
+
+    return this.persistNewAppConfig(args);
+  }
+
+  private createFileWithEmptyConfig() {
+    const emptyConfig: FileAppConfigPresistorConfigSchema = {
+      appRootConfig: {},
+    };
+
+    const writeResult = this.writeJsonFileSafe(
+      this.filePath,
+      JSON.stringify(emptyConfig, null, 2),
+      "utf-8",
+    );
+
+    if (writeResult.isErr()) {
+      return err(writeResult.error);
+    }
+
+    return ok(undefined);
+  }
+
+  async getStripeConfig(args: {
+    channelId: string;
+    saleorApiUrl: string;
+    appId: string;
+  }): Promise<Result<StripeConfig | null, InstanceType<typeof BaseError>>> {
+    const existingConfigResult = this.readExistingAppConfigFromFile();
+
+    if (existingConfigResult.isErr()) {
+      // if file does not exist, create new one with empty config
+      if (existingConfigResult.error instanceof FileAppConfigPresistor.FileReadError) {
+        this.createFileWithEmptyConfig();
+        this.logger.info("File does not exist, creating new one with empty config.");
+
+        return ok(null);
+      }
+
+      return err(existingConfigResult.error);
+    }
+
+    const channelConfig = existingConfigResult.value.appRootConfig[args.channelId];
+
+    if (!channelConfig) {
+      return err(new FileAppConfigPresistor.ConfigNotFoudError("No config found for channelId"));
```

Comment:
1. this should be null, its not error
2. generic errors like not found should be moved to interface file, and implementations are throwing them (sharing)